WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SupportForm/SupportFormViewModel.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+/// Data Source for the Support Request
+///
+public protocol SupportFormMetaDataSource {
+    /// Zendesk Form ID.
+    ///
+    var formID: Int64 { get }
+
+    /// Zendesk tags
+    ///
+    var tags: [String] { get }
+
+    /// Zendesk Custom Fields
+    ///
+    var customFields: [Int64: String] { get }
+}
+
+
+/// View Model for the support form.
+///
+public final class SupportFormViewModel {
+
+    /// Zendesk metadata provider.
+    ///
+    private let dataSource: SupportFormMetaDataSource
+
+    init(dataSource: SupportFormMetaDataSource) {
+        self.dataSource = dataSource
+    }
+
+    /// Submits the support request using the Zendesk Provider.
+    ///
+    func submitSupportRequest(onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        ZendeskProvider.shared.createSupportRequest(formID: dataSource.formID,
+                                                    customFields: dataSource.customFields,
+                                                    tags: dataSource.tags,
+                                                    subject: "Temporary Subject",
+                                                    description: "Temporary Description",
+                                                    onCompletion: onCompletion)
+    }
```

Comment:
I'm not sure... I think this would only be invoked once so probably there is no need?
However, I would keep the suggestion in mind in case we need to chain multiple sync calls!